### PR TITLE
[CAT-103] FIX: Support gzip deserialization

### PR DESCRIPTION
**Pull request by thearchitector**
## Summary

Fixed an issue where `gzip`-compressed files retrieved by the HTTP client would fail deserialization if the inner content type was not unicode-encoded text. It just so happened to work before because we were coincidentally gzipping JSONs.

Breaking change as previously gzip-encoded JSONs fell back to string decoding, thus returned a string which happened to be correct. Results are now `bytes` since we can't know the inner MIME type without external libraries.

## Changelog
- Added `application/gzip` and `application/x-gzip` to the list of known MIME types.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by thearchitector** [comment]
i can make a JIRA if you'd like to track this, not sure where though

File: indico/http/serialization.py [hunk 1]
```diff
@@ -85,5 +85,7 @@ def zip_serialization(content, charset=None):
         "image/png": image_serialization,
         "image/jpg": image_serialization,
         "application/zip": zip_serialization,
+        "application/x-gzip": raw_bytes,
```
Comment (pastap) [inline indico/http/serialization.py:88]: This makes sense - but makes me wonder what the gzip option on `deserialize` is for... @goatrocks ? Any idea where we'd use that?

**Review by pastap — APPROVED** [review]

File: indico/http/serialization.py (hunk 1 above)
Comment (thearchitector) [reply indico/http/serialization.py:88]: it's gets automatically set if the file you're downloading ends in '.gz'. Tbh not great cause now we've kinda created an incorrect that stuff will magically work

Comment (thearchitector) [reply indico/http/serialization.py:88]: occurred to me, do we want to keep doing the `.gz` reading? or should we only decompress if the mime type is correct?

Comment (pastap) [reply indico/http/serialization.py:88]: I would say refer to mime-type and not to file name - but do we ever... set bad mimetypes? not sure... so to be safe, could just `if endswith(.gz) or mimetype == gzip`
that's just my guess tho

Comment (thearchitector) [reply indico/http/serialization.py:88]: nginx sets the mime type, so i doubt we ever send bad ones. we'd have to manually override the response header which i highly doubt we do anywhere

Comment (goatrocks) [reply indico/http/serialization.py:88]: What @thearchitector said is right, it gets set in `client.py` at line 131-138, and assumes that gzip content will end in .gz, but gzip content tends to come with the mime type set not the file extension so if it's enabled on a server this code doesn't work (hence the incorrect assumption)

I would lean towards removing the .gz assumption, but I'm also wary of removing things people may or may not be relying on, but then again I can't think of a situation in which we'd be returning a .gz file..

**Review by goatrocks — APPROVED** [review]

**Comment by thearchitector** [comment]
@goatrocks @pastap yea so adding this def will break people who 1) dont know they are getting a gzip from IPA and 2) who assume gzips will decode to text automatically. the mime type will almost def be correct since nginx is setting it not us -- do we have a final vote on whether or not to remove the file extension logic and rely on mime type?

**Comment by goatrocks** [comment]
> @goatrocks @pastap yea so adding this def will break people who 1) dont know they are getting a gzip from IPA and 2) who assume gzips will decode to text automatically. the mime type will almost def be correct since nginx is setting it not us -- do we have a final vote on whether or not to remove the file extension logic and rely on mime type?

I vote to remove it 👍

**Comment by pastap** [comment]
> > @goatrocks @pastap yea so adding this def will break people who 1) dont know they are getting a gzip from IPA and 2) who assume gzips will decode to text automatically. the mime type will almost def be correct since nginx is setting it not us -- do we have a final vote on whether or not to remove the file extension logic and rely on mime type?
> 
> I vote to remove it

I follow the leader :D

**Pull request merged**